File: canVisitAllRooms/fastestSolution.java
// link: https://leetcode.com/problems/keys-and-rooms/
class Solution {
    private boolean[] visitedRooms = new boolean[rooms.size()];

    public boolean canVisitAllRooms(List<List<Integer>> rooms) {
        moveBetweenRooms(rooms, 0);

        for (boolean isRoomVisited : visitedRooms) {
            if (!isRoomVisited)
                return false;
        }

        return true;
    }

    private void moveBetweenRooms(List<List<Integer>> rooms, int currRoom) {
        if (visitedRooms[currRoom])
            return;
        visitedRooms[currRoom] = true;
        List<Integer> accessibleRooms = rooms.get(currRoom);
        for (int room : accessibleRooms)
            moveBetweenRooms(rooms, room);
    }
}

File: powerOfThree/fastestSolution.cpp
// link: https://leetcode.com/problems/power-of-three/
class Solution {
public:
    bool isPowerOfThree(int n) {
        if(n<=0) return false;        
        // int mxThreshold = 0x7FFFFFFF/3;
        // int mxPoTr = 3;
        // while(mxPoTr < mxThreshold){
        //     mxPoTr = mxPoTr*3;
        // }
        // std::cout << mxPoTr << std::endl;
        int mxPoTr = 1162261467;
        return (mxPoTr%n == 0) ;
    }
};